MELA/test/testME_v2.c

```diff
@@ -1386,6 +1386,292 @@ void testME_VH_JHUGen_Ping(int erg_tev=13, bool useConstants=false, shared_ptr<M
+    mela.selfDZpffcoupl[gHIGGS_Vp_El_left][0]=(-xw*(-1.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Mu_left][0]=(-xw*(-1.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Ta_left][0]=(-xw*(-1.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Dn_left][0]=(-xw*(-1./3.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Up_left][0]=(-xw*(2./3.)+0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Str_left][0]=(-xw*(-1./3.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Chm_left][0]=(-xw*(2./3.)+0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Bot_left][0]=(-xw*(-1./3.)-0.5)*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_El_right][0]=(-xw*(-1.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Mu_right][0]=(-xw*(-1.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Ta_right][0]=(-xw*(-1.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Dn_right][0]=(-xw*(-1./3.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Up_right][0]=(-xw*(2./3.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Str_right][0]=(-xw*(-1./3.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Chm_right][0]=(-xw*(2./3.))*2.;
+    mela.selfDZpffcoupl[gHIGGS_Vp_Bot_right][0]=(-xw*(-1./3.))*2.;
+    mela.selfDM_Zprime=91.1876;
+    mela.selfDGa_Zprime=2.4952;
+    mela.computeP(p0mplus_selfD_CT, useConstants);
+    cout << "p0mplus_selfD_CT: " << p0mplus_selfD_CT << '\n' << endl;
+
+    float p0g1prime2_selfD=0;
+    mela.setProcess(TVar::SelfDefine_spin0, TVar::JHUGen, TVar::ZZGG);
+    mela.selfDHggcoupl[0][gHIGGS_GG_2][0]=1;
+    mela.selfDHzzcoupl[0][gHIGGS_VV_1_PRIME2][0]=1;
+    mela.computeP(p0g1prime2_selfD, useConstants);
+    cout << "p0g1prime2_selfD: " << p0g1prime2_selfD << '\n' << endl;
+
+    float p0hplus_selfD=0;
+    mela.setProcess(TVar::SelfDefine_spin0, TVar::JHUGen, TVar::ZZGG);
+    mela.selfDHggcoupl[0][gHIGGS_GG_2][0]=1;
+    mela.selfDHzzcoupl[0][gHIGGS_VV_2][0]=1;
+    mela.computeP(p0hplus_selfD, useConstants);
+    cout << "p0hplus_selfD: " << p0hplus_selfD << '\n' << endl;
+
+    float p0minus_selfD=0;
+    mela.setProcess(TVar::SelfDefine_spin0, TVar::JHUGen, TVar::ZZGG);
+    mela.selfDHggcoupl[0][gHIGGS_GG_2][0]=1;
+    mela.selfDHzzcoupl[0][gHIGGS_VV_4][0]=1;
+    mela.computeP(p0minus_selfD, useConstants);
+    cout << "p0minus_selfD: " << p0minus_selfD << '\n' << endl;
+
+    float p0gzgs1prime2_selfD=0;
+    mela.setProcess(TVar::SelfDefine_spin0, TVar::JHUGen, TVar::ZZGG);
+    mela.selfDHggcoupl[0][gHIGGS_GG_2][0]=1;
+    mela.selfDHzzcoupl[0][gHIGGS_ZA_1_PRIME2][0]=1;
+    mela.computeP(p0gzgs1prime2_selfD, useConstants);
+    cout << "p0gzgs1prime2_selfD: " << p0gzgs1prime2_selfD << '\n' << endl;
+
+    float p0hpluszgs_selfD=0;
+    mela.setProcess(TVar::SelfDefine_spin0, TVar::JHUGen, TVar::ZZGG);
+    mela.selfDHggcoupl[0][gHIGGS_GG_2][0]=1;
+    mela.selfDHzzcoupl[0][gHIGGS_ZA_2][0]=1;
+    mela.computeP(p0hpluszgs_selfD, useConstants);
+    cout << "p0hpluszgs_selfD: " << p0hpluszgs_selfD << '\n' << endl;
+
+    cout << "*******************************************************" << endl;
+
+    float p2mplus=0;
```

Comment:
No need to duplicate the test. See https://github.com/usarica/HiggsAnalysis-ZZMatrixElement/blob/newVH/MELA/test/testME_v2.c#L5580